Git/config.sublime-syntax

```diff
@@ -0,0 +1,163 @@
+%YAML 1.2
+---
+# Syntax based on documentation here:
+# https://git-scm.com/docs/git-config#_syntax
+
+name: Git Config
+file_extensions:
+  - gitconfig                 # /etc/gitconfig
+  - .gitconfig                # ~/.gitconfig
+first_line_match: ^\[core\]   # .git/config files always start with [core]
+scope: source.gitconfig
+
+variables:
+  section_name: '[\w.-]+(?=[\s\]])'
+  space_or_eol: (?=(?:\s|$))  # makes values on very last line match properly
+  variable_name: '[a-zA-Z][\w-]*'
+
+contexts:
+  prototype:
+    - include: comments
+
+  main:
+    - include: section
+    - include: key-value-pair
+
+  boolean:
+    # 0 and 1 omitted due to mismatches (e.g. "log -1", "HEAD~1"). will be
+    # matched as unquoted strings instead
+    - match: \b(?:yes|true|on|off|no|false){{space_or_eol}}
+      scope: constant.language.gitconfig
+
+  color-attribute:
+    - match: \b(?:no-?)?(?:ul|strike|reverse|italic|dim|bold|blink){{space_or_eol}}
+      scope: support.constant.color-attribute.gitconfig  # TODO: correct scope?
+
+  color-hex:
+    - match: (#)\h{6}
+      captures:
+        0: constant.other.color.rgb-value.gitconfig
+        1: punctuation.definition.constant.gitconfig
+
+  color-named:
+    - match: \b(?:yellow|white|red|normal|magenta|green|cyan|blue|black|auto){{space_or_eol}}
+      scope: support.constant.color.gitconfig
+
+  comments:
+    - match: (#|;).*
```

Comment:
Why not use a push with a meta_scope of comment.line.gitconfig here?

Reply:
That would work too. I'm not sure what the advantage would be. Since they're line comments, I kinda like the added guarantee that the context won't escape the current line.

Reply:
the advantage of @rwols suggestion is that it becomes more likely that you will get correct highlighting if the syntax is ever embedded in another syntax: https://github.com/sublimehq/Packages/issues/757#issuecomment-326882205

Reply:
Done. Should the pop match be `$\r?\n?` to handle Windows line endings, or does it matter?

Reply:
ST normalizes line endings to `\n` internally, so the `\r` would never match, so no :)

Reply:
Awesome! Good to know. Thanks!

Reply:
You should still remove the `.*` part of the match.